bundles/org.openhab.binding.cbus/src/main/java/org/openhab/binding/cbus/handler/CBusCGateHandler.java

```diff
@@ -0,0 +1,433 @@
+                    }
+                    updateGroup(address, state);
+                    break;
+                }
+                case "temperature": {
+                    // For temperature we ignore the state
+                    tokenizer.poll();
+                    @Nullable
+                    String address = tokenizer.poll();
+                    @Nullable
+                    String temp = tokenizer.poll();
+                    updateGroup(address, temp);
+                    break;
+                }
+                case "trigger": {
+                    @Nullable
+                    String command = tokenizer.poll();
+                    @Nullable
+                    String address = tokenizer.poll();
+                    if ("event".equals(command)) {
+                        @Nullable
+                        String level = tokenizer.poll();
+                        updateGroup(address, level);
+                    } else if ("indicatorkill".equals(command)) {
+                        updateGroup(address, "-1");
+                    } else {
+                        logger.warn("Unhandled trigger command {} - status {}", command, status);
+                    }
+                    break;
+                }
+                case "clock": {
+                    @Nullable
+                    String address = "";
+                    @Nullable
+                    String value = "";
+                    @Nullable
+                    String type = tokenizer.poll();
+                    if ("date".equals(type)) {
+                        address = tokenizer.poll() + "/1";
+                        value = tokenizer.poll();
+                    } else if ("time".equals(type)) {
+                        address = tokenizer.poll() + "/0";
+                        value = tokenizer.poll();
+                    } else if (!"request_refresh".equals(type)) {
+                        logger.warn("Received unknown clock event: {}", status);
+                    }
+                    if (value != null && !value.isEmpty()) {
+                        updateGroup(address, value);
+                    }
+                    break;
+                }
+                default: {
+                    LinkedList<String> commentTokenizer = new LinkedList<>(Arrays.asList(contents[1].split("\\s+")));
+                    if ("lighting".equals(commentTokenizer.peek())) {
+                        commentTokenizer.poll();
+                        @Nullable
+                        String commentToken = commentTokenizer.peek();
+                        if ("SyncUpdate".equals(commentToken)) {
+                            commentTokenizer.poll();
+                            @Nullable
+                            String address = commentTokenizer.poll();
+
+                            @Nullable
+                            String level = commentTokenizer.poll();
+                            level.replace("level=", "");
```

Comment:
Strings are immutable, so you need to use the return value.

Reply:
Fixed